packages/web-ui/src/types/breakpoints.ts

```diff
@@ -0,0 +1,2 @@
+export type Breakpoints = 'mobile' | 'tablet' | 'desktop' | 'wide';
+export type Responsive<T> = T | Partial<Record<Breakpoints, T>>;
```

Comment:
what is `T` expected to be here?

Reply:
Any type you want really, enabling you to do...

```tsx
type Props = {
  size?: Responsive<'large' | 'small'>;
  // or
  variant?: Responsive<'solid' | 'outline' | 'ghost'>;
  // or
  spacing?: Responsive<number>;
  // or
  fullWidth?: Responsive<boolean>;
}
```

The `Responsive` type can then properly type responsive objects

```ts
size={{
  mobile: T // 'large' | 'small'
  tablet: T // 'large' | 'small'
  desktop: T // 'large' | 'small'
  wide: T // 'large' | 'small'
}}
```
 
Happy to consider less terse name to replace `T`

Reply:
This also gives us handy autocomplete 🙌🏻 
<img width="533" alt="image" src="https://github.com/utilitywarehouse/design-systems/assets/9257284/a270f286-a328-40d5-837d-7ebe56b75190">

Reply:
is it always expected to be a string?

Reply:
gotcha, makes sense now, nice! 🔥